### DISCUSSION TITLE
Fresh node.js project trying to get vercel work for the first time but when I use vercel dev on terminal i got a spawn cmd.exe error

### DISCUSSION THREAD

**Discussion by isweir** [vercel/vercel#8929] [Help]
vercel dev
Vercel CLI 28.5.3
> Creating initial build
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: spawn cmd.exe ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:285:19)
    at onErrorNT (node:internal/child_process:483:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)      
Emitted 'error' event on ChildProcess instance at:
    at cp.emit (C:\Users\IsWei\AppData\Roaming\npm\node_modules\vercel\node_modules\@ver
cel\build-utils\dist\index.js:17051:29)
    at ChildProcess._handle.onexit (node:internal/child_process:291:12)
    at onErrorNT (node:internal/child_process:483:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {    
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn cmd.exe',
  path: 'cmd.exe',
  spawnargs: [ '/C', 'yarn run dev' ]
}

Node.js v19.1.0
Error: Builder exited with 1 before sending build result

PS: I had added my environment variable paths with C:\Windows\System32

this occures on windows 10.

**Comment by amyegan** [comment]
Version 28.5.5 of Vercel CLI is available now. Do you get the same error after updating to the latest version?

**Reply by isweir** [reply]
C:\Users\IsWei\Documents\GitHub\DiscordCloneBlockChain>vercel --version
Vercel CLI 28.5.5
28.5.5

C:\Users\IsWei\Documents\GitHub\DiscordCloneBlockChain>vercel
Vercel CLI 28.5.5
? Set up and deploy “~\Documents\GitHub\DiscordCloneBlockChain”? [Y/n] y
? Which scope do you want to deploy to? isweir
? Found project “isweir/discordcloneblockchain”. Link to it? [Y/n] n
? Link to different existing project? [Y/n] n
? What’s your project’s name? discordcloneblockchain
? What’s your project’s name? discordcloneblockchain
? In which directory is your code located? ./client
Local settings detected in vercel.json:
No framework detected. Default Project Settings:
- Build Command: `npm run vercel-build` or `npm run build`
- Development Command: None
- Install Command: `yarn install`, `pnpm install`, or `npm install`
- Output Directory: `public` if it exists, or `.`
? Want to modify these settings? [y/N] n
🔗  Linked to isweir/discordcloneblockchain (created .vercel)
🔍  Inspect: https://vercel.com/isweir/discordcloneblockchain/FPAhMxiBVhoSzjzAu3gADztVKHaD [1s]
✅  Production: https://discordcloneblockchain.vercel.app [36s]
📝  Deployed to production. Run `vercel --prod` to overwrite later (https://vercel.link/2F).
💡  To change the domain or build command, go to https://vercel.com/isweir/discordcloneblockchain/settings

C:\Users\IsWei\Documents\GitHub\DiscordCloneBlockChain>vercel dev
Vercel CLI 28.5.5
> Creating initial build
node:events:505
      throw er; // Unhandled 'error' event
      ^

Error: spawn cmd.exe ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess.cp.emit (C:\Users\IsWei\AppData\Roaming\npm\node_modules\vercel\node_modules\@vercel\build-utils\dist\index.js:17051:29)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn cmd.exe',
  path: 'cmd.exe',
  spawnargs: [ '/C', 'yarn run dev' ]
}
Error: Builder exited with 1 before sending build result

C:\Users\IsWei\Documents\GitHub\DiscordCloneBlockChain>node -v
v16.16.0

C:\Users\IsWei\Documents\GitHub\DiscordCloneBlockChain>yarn -v
1.22.19

C:\Users\IsWei\Documents\GitHub\DiscordCloneBlockChain>

**Reply by isweir** [reply]
i was told to have node.js change from v19 to v16 on my email, so that's what i did, and didn't work, now i just update my vercel version and deleted previous vercel files on my project root directory and ran vercel to set it up, then i ran vercel dev and got the same error

**Reply by amyegan** [reply]
Are you able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? I need a closer look at the problem to help with debugging.

**Comment by isweir** [comment]
Didn’t have any example, I created a discordclone folder, inside that folder I created client,server, studio folder, inside the client folder I use box create next app, inside stidio folder I set up sanity, inside server folder I call yarn unit to set up a node server, and finally back in the root discordclone I set up vercel pointing to client folder and run vercel dev, the error occurs. It’s not even a complete project, it’s a new test project that try to run vercel
